pipelines/curate_agentic.py

```
@@ -0,0 +1,613 @@
+    rejected = record.get("rejected")
+    if not isinstance(chosen, dict) or not isinstance(rejected, dict):
+        return False, REASON_SIDES_NOT_OBJECTS
+    present = [goal for goal in preference_goals(record) if goal is not None]
+    if not present:
+        return False, REASON_GOAL_MISSING
+    if len(set(present)) != 1:
+        return False, REASON_GOAL_DIVERGES
+    return True, None
+
+
+def _basis_missing(step: Any) -> bool:
```

Ruff (0.16.1)

[warning] Dynamically typed expressions (typing.Any) are disallowed in `step`

(ANN401)